### DISCUSSION TITLE
Do I need to have jsdom as devDependency when changing environment option to "jsdom"?

### DISCUSSION THREAD

**Discussion by MantasSutkevicius** [vitest-dev/vitest#1077] [Q&A] [answered]
If I change the `environment` property in `vitest.config.ts` to `jsdom` and run the test, it says that jsdom is not found and it asks if I would like to install it within my own dependencies. Is this expected? Because it seems like sometimes (I've tried this in various different places, deleting node_modules and reinstalling packages etc.) it just works without me having this jsdom package.

**Comment by sheremet-va** [comment, marked as answer]
jsdom and happy-dom should be installed separately depending on which one you choose.

sometimes some other dependency can install it, i guess - and that's why it's working
